query_executor.go

```diff
@@ -45,9 +45,11 @@ func (q *queryExecutor) checkRetryPolicy(rq ExecutableQuery, err error) (RetryTy
 	if p == nil {
 		return Rethrow, err
 	}
-	if p.Attempt(rq) {
+	if p.Attempt(rq, host) {
 		return p.GetRetryType(err), nil
 	}
+
+	// p.Attempt is false, should not retry anything
 	return p.GetRetryType(err), err
```

Comment:
Well, that's also a problem. If the retryPolicy is implemented similarly to the new testRetryPolicy, then when those attempts are done it returns `RetryNextHost`, not Rethrow. If you ignore that, then you're throwing an error instead of trying the next host.

That's why I ended up accepting the need to call it here, but check it in the queryExecutor. I don't like both solutions, but at least the current one seems to be working correctly.

Reply:
Would you mind if I fix this in my branch because I'm working on #1161 and it touches the same code path again in the query executor?